src/test/java/uk/gov/pay/directdebit/mandate/services/MandateSearchServiceTest.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.pay.directdebit.mandate.services;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.pay.directdebit.mandate.dao.MandateSearchDao;
+import uk.gov.pay.directdebit.mandate.params.MandateSearchParams;
+
+import java.util.List;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.pay.directdebit.mandate.fixtures.MandateFixture.aMandateFixture;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class MandateSearchServiceTest {
+
+    @Mock
+    private MandateSearchDao mockMandateSearchDao;
+
+    private MandateSearchService mandateSearchService;
+
+    @Before
+    public void setUp() {
+        mandateSearchService = new MandateSearchService(mockMandateSearchDao);
+    }
+
+    @Test
+    public void shouldReturnOneMandateForPageAndTotalOfTwoMatchingMandates() {
+        var params = MandateSearchParams.MandateSearchParamsBuilder.aMandateSearchParams()
+                .withName("aName")
+                .build();
+
+        var mandateListToReturn = List.of(
+                aMandateFixture().withServiceReference("expectedReference").toEntity()
+        );
```

Comment:
It’s a little confusing that there are ostensibly two results but this list only has one

Reply:
I've tried to address that in the method names with `getTotalMatchingMandates` and `getMandatesForRequestedPage`. I deliberately used different values because it's reasonable that the total count is greater than the mandates required just for the requested page.